File: app/src/main/java/com/vwoom/timelapsegallery/database/dao/ProjectScheduleDao.java
package com.vwoom.timelapsegallery.database.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.vwoom.timelapsegallery.database.entry.ProjectScheduleEntry;
import com.vwoom.timelapsegallery.database.entry.ProjectTagEntry;

import java.util.List;

@Dao
public interface ProjectScheduleDao {

    @Query ("SELECT * FROM project_schedule WHERE project_id = :projectId")
    ProjectScheduleEntry loadScheduleByProjectId(long projectId);

    @Insert
    void insertProjectSchedule(ProjectScheduleEntry projectScheduleEntry);

    @Delete
    void deleteProjectSchedule(ProjectScheduleEntry projectScheduleEntry);

    @Update
    void updateProjectSchedule(ProjectScheduleEntry projectScheduleEntry);

}

File: app/src/main/java/com/vwoom/timelapsegallery/database/dao/ProjectScheduleDao.kt
package com.vwoom.timelapsegallery.database.dao

import androidx.lifecycle.LiveData
import androidx.room.*
import com.vwoom.timelapsegallery.database.entry.ProjectScheduleEntry

@Dao
interface ProjectScheduleDao {
    @Query("SELECT * FROM project_schedule")
    fun loadProjectSchedules(): LiveData<List<ProjectScheduleEntry?>?>?

    @Query("SELECT * FROM project_schedule WHERE project_id = :projectId")
    fun loadScheduleByProjectId(projectId: Long): ProjectScheduleEntry?

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertProjectSchedule(projectScheduleEntry: ProjectScheduleEntry?)

    @Delete
    fun deleteProjectSchedule(projectScheduleEntry: ProjectScheduleEntry?)

    @Update
    fun updateProjectSchedule(projectScheduleEntry: ProjectScheduleEntry?)
}